src/pages/revision/[id].tsx

```diff
@@ -0,0 +1,215 @@
+        toc.length
+      }`
+
+      // TODO: Find out why this is happening
+      // if the last item in toc is same as current item, remove the last item
+      // to avoid duplicate items
+      if (toc[toc.length - 1]?.title === children[0]) {
+        toc.pop()
+      }
+      toc.push({
+        level,
+        id,
+        title: children[0],
+      })
+      return React.createElement(props.node.tagName, { id }, children)
+    }
+    return React.createElement(props.node.tagName, props, children)
+  }
+
+  const addWikiPreview = ({
+    children,
+    ...props
+  }: React.PropsWithChildren<
+    ComponentPropsWithoutRef<'a'> & ReactMarkdownProps
+  >) => {
+    // TODO: Make more specific regex
+    const wikiLinkRecognizer = /.*\/wiki\/(.*)/
+    const wikiSlug = props?.href?.match(wikiLinkRecognizer)?.[1]
+
+    // Checks if the link is a wiki link
+    const isChildrenPresent =
+      children && typeof children[0] === 'string' && children[0].length > 0
+    const isWikiSlugPresent = wikiSlug && wikiSlug.length > 0
+
+    // render special hover component if the link is a wiki link
+    if (isChildrenPresent && isWikiSlugPresent && props.href) {
+      return (
+        <WikiPreviewHover
+          text={children[0] as string}
+          href={props.href}
+          slug={wikiSlug}
+        />
+      )
+    }
+    return React.createElement(props.node.tagName, props, children)
+  }
+  /* eslint-enable react/prop-types */
+
+  return (
+    <>
+      {wiki && (
+        <NextSeo
+          title={wiki.content[0].title}
+          openGraph={{
+            title: wiki.content[0].title,
+            description: getWikiSummary(wiki.content[0]),
+            images: [
+              {
+                url: getWikiImageUrl(wiki.content[0]),
+              },
+            ],
+          }}
+        />
+      )}
+
+      <main>
+        {!error && (router.isFallback || isLoading) ? (
+          <Flex justify="center" align="center" h="50vh">
+            <Spinner size="xl" />
+          </Flex>
+        ) : (
+          <VStack mt={-2}>
+            <Flex
+              flexDir={{ base: 'column', lg: 'row' }}
+              justify="center"
+              align="center"
+              gap={2}
+              bgColor="red.200"
+              _dark={{ bgColor: 'red.500' }}
+              w="100%"
+              p={2}
+            >
+              <Text textAlign="center">
+                You are seeing an older version of this wiki.
+              </Text>
+              <Link href={`/wiki/${wiki?.content[0].id}`} passHref>
+                <Button
+                  as="a"
+                  maxW="120px"
+                  variant="solid"
+                  bgColor="dimColor"
+                  _hover={{ bgColor: 'dimColor', textDecoration: 'underline' }}
+                  _focus={{ bgColor: 'dimColor', textDecoration: 'underline' }}
+                  _active={{ bgColor: 'dimColor', textDecoration: 'underline' }}
```

Comment:
We could do this, to avoid repetition
```suggestion
                  sx={{
                    '&:hover, &:focus, &:active': {
                      bgColor: 'dimColor',
                      textDecoration: 'underline',
                    },
                  }}